ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -706,10 +709,27 @@ private Page<NamedBlobRecord> run_list_v2(String accountName, String containerNa
         int resultIndex = 0;
         while (resultSet.next()) {
           String blobName = resultSet.getString(1);
-          if (resultIndex++ == maxKeysValue) {
+          if (resultIndex == maxKeysValue) {
             nextContinuationToken = blobName;
             break;
           }
+
+          if (groupDirectories) {
+            // Extract the portion after the prefix and before the next '/'
+            String remainingPath = blobName.substring(blobNamePrefix == null ? 0 : blobNamePrefix.length());
+            remainingPath = remainingPath.startsWith("/") ? remainingPath.substring(1) : remainingPath;
```

Comment:
this is not right, if the prefix is "abc" and this blobname is "abc/efg/hij", and the delimiter is "/", then the we should "abc/" as common prefix, not "abc/efg/".

Reply:
Sure! Corrected it